tests/test_vert_coords.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import pytest
+
+from pyaerocom.griddeddata import GriddedData
+from pyaerocom.vert_coords import *
+from tests.conftest import lustre_unavail
+from tests.fixtures.tm5 import data_tm5
+
+
+@pytest.fixture
+def alt(data_tm5: GriddedData) -> AltitudeAccess:
+    return AltitudeAccess(data_tm5)
+
+
+@pytest.fixture
+def sin_wave() -> np.ndarray:
+    return np.sin(np.arange(1000))
+
+
+def test_atmosphere_sigma_coordinate_to_pressure_exceptions():
+    with pytest.raises(ValueError):
+        atmosphere_sigma_coordinate_to_pressure(1.0, 0, "")
+
+
+@pytest.mark.parametrize(
+    "sigma,ps,ptop",
+    (
+        pytest.param(1.0, 0, 0.0, id="ptop-is-float"),
+        pytest.param(1.0, 0, np.ones(10), id="ptop-is-ndarray"),
+    ),
+)
+def test_atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop):
+    result = atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop)
+
+    assert type(result) == type(ptop)
+    if isinstance(result, np.ndarray):
+        assert len(result) == len(ptop)
+
+
+def test_atmosphere_hybrid_sigma_pressure_coordinate_to_pressure_exceptions():
+    with pytest.raises(ValueError):
+        atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(np.ones(10), np.ones(5), 1, 0)
+
+
+@pytest.mark.parametrize(
+    "a,b,ps,p0",
+    (
+        pytest.param(np.ones(10), np.ones(10), 1, None, id="test-p0-is-none"),
+        pytest.param(np.ones(10), np.ones(10), 1, 1, id="test-p0-is-not-none"),
+    ),
+)
+def test_atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(
+    a: np.ndarray, b: np.ndarray, ps: float, p0: float | None
+):
+    result = atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(a, b, ps, p0)
+
+    assert len(a) == len(b) == len(result)
+
+
+def test_geopotentialheight2altitude():
+    with pytest.raises(NotImplementedError):
+        geopotentialheight2altitude(5)
+
+
+def test_VerticalCoordinate_exceptions():
+    # Unsupported variable name during initialization.
+    with pytest.raises(ValueError):
+        VerticalCoordinate("jtlkjhsklrg")
+
+    # Attempt calculating pressure for unsupported variable.
+    vert = VerticalCoordinate("asc")
+    with pytest.raises(CoordinateNameError):
+        vert.calc_pressure(np.ones(5))
```

Comment:
please split this test or use `pytest.mark.parameterize`.
As written, a failure on the 1st assert will shadow a failure on the second.

Reply:
Done